### PR TITLE
clean up API for starting Flutter commands

### PR DESCRIPTION
**Pull request by skybrian**
Introduced FlutterCommand and moved the low-level details of launching
processes there.

Also, somewhat better error reporting in a few places.

### CODE REVIEW

File: src/io/flutter/ProjectOpenActivity.java [hunk 1]
```diff
@@ -84,7 +84,7 @@ private boolean downloadDependencies(@NotNull Project project, @NotNull PubRoot
       // We have a .packages file, but need to run a flutter command to download the Dart SDK.
       final boolean ok = sdk.sync(project);
       if (!ok) {
-        LOG.warn("failed to sync flutter SDK");
+        FlutterMessages.showError("Error opening", "failed to download Dart SDK for Flutter");
```
Comment (devoncarew) [inline src/io/flutter/ProjectOpenActivity.java]: `Error Opening Project`?, and we should capitalize `Failed`.

**Comment by devoncarew** [comment]
lgtm

(I suspect we should wait to land this until M13.1 is out)

**Comment by skybrian** [comment]
Ok to commit?

**Comment by pq** [comment]
Yep.  Bombs away!

💣

**Pull request merged**